src/web/js/dashboard/TeacherDashboard.js

```diff
@@ -0,0 +1,202 @@
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+
+import GoogleAPI from './GoogleAPI.js';
+import '../../css/dashboard/index.css';
+
+import 'react-mdl/extra/material.css';
+import 'react-mdl/extra/material.js';
+import { Layout, Header, Drawer, Navigation, Content, Button, HeaderRow, HeaderTabs, Tab, Snackbar } from 'react-mdl';
+
+import ClassList from './ClassList';
+import StudentList from './StudentList';
+import AssignmentList from './AssignmentList';
+
+class TeacherDashboard extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      signedIn: false,
+      classes: {},
+      activeTab: 0,
+      activeClassId: false,
+      activeClass: '',
+      studentsInClass: [],
+      assignmentsInClass: [],
+      studentSpinnerActive: true,
+      classSpinnerActive: true,
+      assignmentSpinnerActive: true,
+      isSnackbarActive: false,
+      snackbarText: ''
+    };
+
+    this.api = new GoogleAPI();
+    this.api.load().then(() => {
+      this.handlePageLoad();
+    });
+  }
+
+  componentDidUpdate = (prevProps, prevState) => {
+    if (this.state.studentsInClass != prevState.studentsInClass) {
+      this.setState({studentSpinnerActive: false});
+    }
+    if (this.state.classes != prevState.classes) {
+      this.setState({classSpinnerActive: false});
+    }
+    if (this.state.assignmentsInClass != prevState.assignmentsInClass) {
+      this.setState({assignmentSpinnerActive: false});
+    }
+  }
+
+  setSnackBarMessage = (message) => {
+    this.setState({
+      isSnackbarActive: true,
+      snackbarText: message
+    })
+  }
+
+  handleTimeoutSnackbar = () =>{
+    this.setState({ isSnackbarActive: false });
+  }
+
+  handleClickActionSnackbar = () => {
+    this.setState({ isSnackbarActive: false });
+  }
+
+  handleSignInClick = () => {
+    this.api.signIn().then(() => {
+      this.handlePageLoad();
+    });
+  }
+
+  handleSignOutClick = () => {
+    this.setState({signedIn: false});
+    window.location.replace('/logout');
+  }
+
+  handlePageLoad = () => {
+    this.setState({signedIn: true});
+    this.api.initializePyretData().then(() => {
+      this.refreshState();
+    });
+  }
+
+  refreshState = () => {
+    this.api.getAllClasses().then((resp) => {
+      const classes = resp;
+      this.setState({
+        classes: classes,
+        activeClassId: 'class' + Object.keys(classes)[0] || '',
+        activeClass: Object.keys(classes)[0] || false
+      }, () => {
+        this.refreshInnerState();
+      })
+    });
+  }
+
+  refreshInnerState = () => {
+    if (this.state.activeClass) {
+      this.api.getStudentsInClass(this.state.activeClass).then(resp => {
```

Comment:
More promises that fail silently? (I will check the API later to see if it sets some global error notification)